lib/ReactViews/ExplorerWindow/Tabs/DataCatalogTab.tsx

```diff
@@ -0,0 +1,92 @@
+import { runInAction } from "mobx";
+import { observer } from "mobx-react";
+import { useTranslation } from "react-i18next";
+import type CatalogMemberMixin from "../../../ModelMixins/CatalogMemberMixin";
+import Box from "../../../Styled/Box";
+import { useViewState } from "../../Context";
+import DataCatalog from "../../DataCatalog/DataCatalog";
+import DataPreview from "../../Preview/DataPreview";
+import Breadcrumbs from "../../Search/Breadcrumbs";
+import SearchBox, { DEBOUNCE_INTERVAL } from "../../Search/SearchBox.jsx";
+import Styles from "./data-catalog-tab.scss";
+
+interface DataCatalogTabProps {
+  items?: unknown[];
+  searchPlaceholder?: string;
+  onActionButtonClicked?: (item: CatalogMemberMixin.Instance) => void;
+}
+
+const DataCatalogTab = observer(function DataCatalogTab(
+  props: DataCatalogTabProps
+) {
+  const viewState = useViewState();
+  const { t } = useTranslation();
+
+  const {
+    searchState,
+    previewedItem: previewed,
+    breadcrumbsShown: showBreadcrumbs,
+    terria
+  } = viewState;
+
+  const searchPlaceholder =
+    props.searchPlaceholder || t("addData.searchPlaceholder");
+
+  const changeSearchText = (newText: string) => {
+    runInAction(() => {
+      viewState.searchState.catalogSearchText = newText;
+    });
+  };
+
+  const search = () => {
+    viewState.searchState.searchCatalog();
+  };
+
+  return (
+    <div className={Styles.root}>
+      <Box fullHeight column>
+        <Box fullHeight overflow="hidden">
+          <Box className={Styles.dataExplorer} styledWidth="40%">
+            {searchState.catalogSearchProvider && (
+              <SearchBox
+                searchText={searchState.catalogSearchText}
+                onSearchTextChanged={changeSearchText}
+                onDoSearch={() => search()}
+                placeholder={searchPlaceholder}
+                debounceDuration={
+                  terria.catalogReferencesLoaded
+                    ? (searchState.catalogSearchProvider as any)
+                        .debounceDurationOnceLoaded
+                    : DEBOUNCE_INTERVAL
+                }
```

Comment:
Possible error: Previously, if  either `terria.catalogReferencesLoaded` is `false` or if `searchState.catalogSearchProvider` is `undefined` the duration defaults to `DEBOUNCE_INTERVAL`. But now if `searchState.catalogSearchProvider` is `undefined` it could result in a type error?

Also looking at the `any` cast, I'm thinking whether we can hardcode `debounceDurationOnceLoaded` within `DataCatalogTab` component. I don't see a reason why it needs to be configurable at the `CatalogSearchProvider` level.

Reply:
`debounceDurationOnceLoaded` was added in #4803 don't know the exact reason why it was added. I have improved type cast, but I also don't see why we need that we can just hardcode the value